pkg/api/v1/types.go

```diff
@@ -1469,10 +1479,18 @@ type Container struct {
 	// Optional: Path at which the file to which the container's termination message
 	// will be written is mounted into the container's filesystem.
 	// Message written is intended to be brief final status, such as an assertion failure message.
+	// Will be truncated by the node if too large.
```

Comment:
I left this vague in the doc until we had a discussion here about max size.

The current value is 4096 bytes per container.  That's potentially a lot.  I think 20-40 lines of backtrace in a normal log file would provide a fair amount of context, which is probably ~1600 bytes on average.  For termination files, we would want to accept more, since every byte belongs to something the user intends to send.  Overall, we can't accept more than probably 5k-20k per pod (the upper end of that range might be 5-10x the average pod size).

How about for simplicity sake we apply the following limits:

1. For reading logs we read 2k bytes or 40 lines, whichever is smaller (from the end of the logs)
2. For termination logs we allow up to 4k bytes (from the beginning of the file)
3. If all container messages take up more than 12k bytes, we truncate messages proportionally from the beginning of the message (rather than doing something more complicated which mixes beginning and end truncation).